frontend/test/resolution/testTuples.cpp

```diff
@@ -491,26 +491,152 @@ static void test13() {
   assert(vt == tt);
 }
 
+// TupleDecl formals
 static void test14() {
   printf("test14\n");
-  Context ctx;
-  Context* context = &ctx;
 
-  auto qt = resolveTypeOfXInit(context,
-                R""""(
-                  proc f(x: int, (y, z): (real, int)) { return (x, y, z); }
-                  var x = f( 1, (2.0, 3) );
-                )"""");
+  {
+    Context ctx;
+    Context* context = &ctx;
+    ErrorGuard guard(context);
 
-  assert(qt.kind() == QualifiedType::CONST_VAR);
-  assert(qt.type()->isTupleType());
-  auto tt = qt.type()->toTupleType();
+    auto qt = resolveTypeOfXInit(context,
+                  R""""(
+                    proc f(x: int, (y, z): (real, int)) { return (x, y, z); }
+                    var x = f( 1, (2.0, 3) );
+                  )"""");
+
+    assert(qt.kind() == QualifiedType::CONST_VAR);
+    assert(qt.type()->isTupleType());
+    auto tt = qt.type()->toTupleType();
+
+    assert(tt->numElements() == 3);
+    assert(!tt->isStarTuple());
+    assert(tt->elementType(0).type()->isIntType());
+    assert(tt->elementType(1).type()->isRealType());
+    assert(tt->elementType(2).type()->isIntType());
+  }
 
-  assert(tt->numElements() == 3);
-  assert(!tt->isStarTuple());
-  assert(tt->elementType(0).type()->isIntType());
-  assert(tt->elementType(1).type()->isRealType());
-  assert(tt->elementType(2).type()->isIntType());
+  {
+    Context ctx;
+    Context* context = &ctx;
+    ErrorGuard guard(context);
+
+    // Using a homogeneous tuple type expression
+    auto vars = resolveTypesOfVariables(context,
+                  R""""(
+                    proc foo((x, y, z): 3*int) {
+                      return x;
+                    }
+
+                    var three = (1, 2, 3);
+                    var retOne = foo(three);
+
+                    var a, b, c : int;
+
+                    var retTwo = foo((a, b, c));
+                  )"""",
+                  {"retOne", "retTwo"});
+    assert(!vars["retOne"].isUnknown());
```

Comment:
Can we also assert that they're `TupleType`s here?

Reply:
These return values should just be integers, but I definitely should have been checking for that instead of Unknown.